tests/testsuite/registry_overlay.rs

```diff
@@ -0,0 +1,156 @@
+//! Tests for local-registry sources.
+
+use cargo_test_support::registry::{Package, RegistryBuilder, TestRegistry};
+use cargo_test_support::{project, Project};
+
+fn setup() -> (TestRegistry, String) {
+    let alt = RegistryBuilder::new().alternative().build();
+    (
+        RegistryBuilder::new().http_index().build(),
+        alt.index_url()
+            .to_file_path()
+            .unwrap()
+            .into_os_string()
+            .into_string()
+            .unwrap(),
+    )
+}
+
+fn simple_project() -> Project {
```

Comment:
Let's avoid using this.  This reduces "duplication" but that is false code duplication.  They have the appearance of being the same but they do not share requirements.  Instead it affects the code locality (the ability to read tightly coupled code).  I'm trying to double check what these tests are doing and jumping back and forth is making this more difficult.